engine/execution/computation/manager.go

```diff
@@ -302,22 +287,8 @@ func (e *Manager) ComputeBlock(
 		Hex("block_id", logging.Entity(block.Block)).
 		Msg("block result computed / derived data cache updated")
 
-	if uploadEnabled {
-		var group errgroup.Group
-
-		for _, uploader := range e.uploaders {
-			uploader := uploader
-
-			group.Go(func() error {
-				span, _ := e.tracer.StartSpanFromContext(ctx, trace.EXEUploadCollections)
-				defer span.End()
-
-				return uploader.Upload(result)
-			})
-		}
-
-		err = group.Wait()
-
+	if e.uploader.Enabled() {
+		err := e.uploader.Upload(ctx, result)
```

Comment:
can we move the uploader.Upload to ingestion engine right after it's calling [manager.ComputeBlock()](https://github.com/onflow/flow-go/blob/master/engine/execution/ingestion/engine.go#L618)?

That way, manager doesn't need to depend on uploader. Ingestion engine depends on the uploader anyway, so we can just let ingestion engine take care of the upload.

If we think about it, the result upload has nothing to do with computation. It simply upload the result to cloud.

If we move the uploader to ingestion engine, then the uploader.Manager module could also move under ingestion package, which is right now in the manager package.

Reply:
updated in 7b82842 and bd39327